distribution/docs/src/main/jdocs/content/_developingComponents/developing-plugins-contents.adoc

```diff
@@ -0,0 +1,403 @@
+:title:  Developing Catalog Plugins
+:type: developingComponent
+:status: published
+:link: _developing_catalog_plugins
+:summary: Creating a custom catalog plugin.
+
+Plugins extend the functionality of the Catalog Framework by performing actions at specified times during a transaction. 
+Plugin interfaces are located in the Catalog Core API.
+By implementing a plugin interface, actions can be performed at the desired time. 
+
+[NOTE]
+====
+Catalog plugin interfaces are located in the catalog core API.
+====
+
+The following types of plugins can be created:
+
+.Plugin Interfaces
+[cols="1,2m,1", options="header"]
+|===
+
+|Plugin Type
+|Plugin Interface
+|Invocation Order
+
+|<<_pre_authorization_plugins,Pre-Authorization>>
+|ddf.catalog.plugin.PreAuthorizationPlugin
+|Before any security rules are implemented.
+
+|<<_policy_plugins,Policy>>
+|ddf.catalog.plugin.PolicyPlugin
+|After pre-authorization plugins, but before other catalog plugins to establish the policy for requests/responses.
+
+|<<_access_plugins,Access>>
+|ddf.catalog.plugin.AccessPlugin
+|Directly after any policy plugins
+
+|<<_pre_ingest_plugins,Pre-Ingest>>
+|ddf.catalog.plugin.PreIngestPlugin
+|Before the Create/Update/Delete method is sent to the `CatalogProvider`.
```

Comment:
Yes. I don't see any real reason to go with the interface name, so changed to `Catalog Provider`